src/crystal/system/unix/file.cr

```diff
@@ -105,6 +105,25 @@ module Crystal::System::File
     ret
   end
 
+  def self.readlink(path) : String
+    buf = Bytes.new 256
+    3.times do
+      ret = LibC.readlink(path, buf, buf.bytesize)
+      if ret == -1
+        raise Errno.new("readlink")
+      elsif ret == buf.bytesize
+        buf = Bytes.new(buf.bytesize * 4)
```

Comment:
On the 3rd pass, if the buffer is still too small, it will allocate another memory chunk that will never be used.